### PR TITLE
Make all videos playable on fronts

### PR DESCRIPTION
**Pull request by johnduffell**
At present we only make the first 4 (or 1 for tag pages) videos playable on the fronts (for mobile and desktop).  This is because if there are lots, it tries to preload them all, and also breaks mobile.

So this change is to make it a placeholder image with an overlayed play button, and when you hit the button it replaces the placeholder with the real video and starts it playing in place.

There are a few things I haven't done - the hover for the play button isn't implemented yet.  Also the Third container size still isn't playable, but I don't think making it playable would be good in all cases - at mobile breakpoint the first third should be playable but the other two shouldn't:
thirds-transformation
I'd suggest considering that as a separate issue.

Also there are a few comments in the code for things I wasn't 100% sure on.  I will take them out before I merge.

### CODE REVIEW

File: common/app/cards/CardType.scala [hunk 1]
```diff
@@ -4,6 +4,8 @@ sealed trait CardType {
   val cssClassName: String
 
   def showVideoPlayer = this match {
+      // TODO check this is really correct - only should be playable if they're big enough - Third could be playable on desktop, but at mobile it renders them wide, narrow, narrow above each other
+      // so only the top one makes sense to be playable
```
Comment (robertberry) [inline common/app/cards/CardType.scala]: Mobile + Tablet card sizes can be mixed willy nilly (although there are probably only a small set of actual combinations). This parameter is to say whether a video player appears at *any* breakpoint. If it does, then we needed previously to show the video tag.

File: common/app/views/fragments/items/facia_cards/contentCard.scala.html [hunk 2]
```diff
@@ -32,30 +32,33 @@
...
-                @fallback.map { fallbackImage =>
-                    <div class="fc-item__video-fallback">
-                        @image(
-                            fallbackImage.imageContainer,
-                            inlineImage = containerIndex == 0 && index < 4
-                        )
-                    </div>
-                }
-                }
+                        @fallback.map { fallbackImage => <!-- FIXME what if no fallback image? could it happen? -->
```
Comment (robertberry) [inline common/app/views/fragments/items/facia_cards/contentCard.scala.html]: I think it probably doesn't happen, but it might be worth checking out.

**Comment by robertberry** [comment]
Somebody more familiar with the JavaScript ought to check this out, but the Scala looks good to me. :+1:

File: dev-build/conf/application-logger.xml [hunk 3]
```diff
@@ -19,8 +19,20 @@
...
+            <onMismatch>DENY</onMismatch>
+        </filter>
+        <encoder>
+            <pattern>%date [%thread] %-5level %logger{36} - %msg%n%xException</pattern>
+        </encoder>
+    </appender>
+
     <root level="INFO">
         <appender-ref ref="LOGFILE"/>
+        <appender-ref ref="Console"/>
```
Comment (phamann) [inline dev-build/conf/application-logger.xml]: Why is this file in this PR?

**Comment by phamann** [comment]
You've got a merge conflict

**Comment by phamann** [comment]
This PR is much easier to read if you ignore white space changes via [?w=0](https://github.com/guardian/frontend/pull/7891)

**Comment by phamann** [comment]
:+1:

File: dev-build/conf/application-logger.xml (hunk 3 above)
Comment (johnduffell) [inline dev-build/conf/application-logger.xml]: forgot to move it! =)

File: common/app/cards/CardType.scala (hunk 1 above)
Comment (johnduffell) [inline common/app/cards/CardType.scala]: ok thanks, as discussed with you and @crifmulholland I will show the play button at third size, but use CSS rules to hide them when they go down to tiny size at mobile breakpoint - I will do this in a new PR.

File: static/src/stylesheets/module/_media.scss [hunk 4]
```diff
@@ -36,3 +36,9 @@ Nicole Sullivan's OOCSS CSS markup extraction pattern.
 .media__img img {
     display: block;
 }
+
+.media__container--hidden { display: none; }
+.media__placeholder--active { display: block; }
+
+.media__container--active { display: block; }
+.media__placeholder--hidden { display: none; }
```
Comment (sammorrisdesign) [inline static/src/stylesheets/module/_media.scss]: Could you use `u-h` class to hide and show these instead?

Comment (johnduffell) [inline static/src/stylesheets/module/_media.scss]: Thanks good point @sammorrisdesign and I did consider that but it seems to be designed to hide it visibly but keep it on the page (e.g. for screen readers)  I'm not sure if that's what I need as I am substituting two equivalents completely.  What do you think?  When would we use that over display: none in general?

**Comment by johnduffell** [comment]
opening a new pull request as it's changed a lot

**Comment by gidsg** [comment]
Does this play nice with screenreaders e.g. Voiceover on iOS?

**Pull request closed without merging**